pkg/utils/kubernetes/utils.go

```diff
@@ -0,0 +1,37 @@
+package kubernetes
+
+import (
+	"errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func CustomResourceDefinitionExists(gvk schema.GroupVersionKind, cfg ...*rest.Config) (bool, error) {
+	var err error
+	if len(cfg) > 0 && cfg[0] == nil {
+		cfg = cfg[0:0]
+	}
+	if len(cfg) == 0 {
+		tcfg, err := config.GetConfig()
+		if err != nil {
+			return false, err
+		}
+		cfg = append(cfg, tcfg)
+	}
+	if err != nil {
+		return false, err
+	}
+	client, err := discovery.NewDiscoveryClientForConfig(cfg[0])
```

Comment:
What if there is an error at this point?

Reply:
fixed